inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -0,0 +1,374 @@
+    std::string name = n->get_type_name();
+    NGRAPH_CHECK(name != "GenericIE", "Type of ", n->get_name(), " (",
+                 n->get_type_name(), ") ", "is not supported.");
+
+    const std::map<std::string, std::string> translator = {
+        {"Constant", "Const"}};
+    if (translator.count(name) > 0) {
+        name = translator.at(name);
+    }
+    return name;
+}
+
+std::string get_output_precision_name(ngraph::Output<Node>& o) {
+    auto elem_type = o.get_element_type();
+    switch (elem_type) {
+    case ::ngraph::element::Type_t::undefined:
+        return "UNSPECIFIED";
+    case ::ngraph::element::Type_t::f16:
+        return "FP16";
+    case ::ngraph::element::Type_t::f32:
+        return "FP32";
+    case ::ngraph::element::Type_t::bf16:
+        return "BF16";
+    case ::ngraph::element::Type_t::i8:
+        return "I8";
+    case ::ngraph::element::Type_t::i16:
+        return "I16";
+    case ::ngraph::element::Type_t::i32:
+        return "I32";
+    case ::ngraph::element::Type_t::i64:
+        return "I64";
+    case ::ngraph::element::Type_t::u8:
+        return "U8";
+    case ::ngraph::element::Type_t::u16:
+        return "U16";
+    case ::ngraph::element::Type_t::u32:
+        return "U32";
+    case ::ngraph::element::Type_t::u64:
+        return "U64";
+    case ::ngraph::element::Type_t::u1:
+        return "BIN";
+    case ::ngraph::element::Type_t::boolean:
+        return "BOOL";
+    default:
+        NGRAPH_CHECK(false, "Incorrect precision: ", elem_type.get_type_name());
+        return "";
+    }
+}
+
+std::string generate_unique_name(
+    const std::unordered_set<std::string>& unique_names, std::string base_name,
+    int suffix) {
+    std::string new_name = base_name + std::to_string(suffix);
+    if (unique_names.find(new_name) == unique_names.end()) {
+        return new_name;
+    } else {
+        suffix++;
+        return generate_unique_name(unique_names, base_name, suffix);
+    }
+}
+
+// TODO: remove when CNNNetwork will be supporting not-unique names
+std::string get_node_unique_name(std::unordered_set<std::string>& unique_names,
+                                 const ngraph::Node* n) {
+    std::string name = n->get_friendly_name();
+    if (unique_names.find(name) != unique_names.end()) {
+        name = generate_unique_name(unique_names, name, 0);
+    }
+    unique_names.insert(name);
+    return name;
+}
+
+void ngfunction_2_irv10(pugi::xml_document& doc, std::vector<uint8_t>& bin,
+                        const ngraph::Function& f) {
+    pugi::xml_node netXml = doc.append_child("net");
+    netXml.append_attribute("name").set_value(f.get_friendly_name().c_str());
+    netXml.append_attribute("version").set_value("10");
+    pugi::xml_node layers = netXml.append_child("layers");
+
+    const std::map<ngraph::Node*, int> layer_ids = create_layer_ids(f);
+    std::unordered_set<std::string> unique_names;
+
+    for (auto n : f.get_ordered_ops()) {
```

Comment:
nit: please use const reference, we don't modify nodes in this code.
```suggestion
    for (const auto& n : f.get_ordered_ops()) {
```

Reply:
Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/09fbad0067999b153c2c78be728aafef935d25aa)